Orange/widgets/utils/signals.py

```diff
@@ -0,0 +1,242 @@
+import copy
+
+from Orange.canvas.registry.description import InputSignal, OutputSignal
+
+
+class _Signal:
+    @staticmethod
+    def get_flags(multiple, default, explicit, dynamic):
+        """Compute flags from arguments"""
+        from Orange.canvas.registry.description import \
+            Single, Multiple, Default, NonDefault, Explicit, Dynamic
+        return (Multiple if multiple else Single) | \
+                (Default if default else NonDefault) | \
+                (explicit and Explicit) | \
+                (dynamic and Dynamic)
+
+
+class Input(InputSignal, _Signal):
+    """
+    Description of an input signal.
+
+    The class is used to declare input signals for a widget as follows
+    (the example is taken from the widget Test & Score)::
+
+        class Inputs:
```

Comment:
This was why I use singular with messages. I'm kind of undecided, you vote for singular (yes?)... @astaric?